ark/mibi/tiling_utils_test.py

```diff
@@ -465,125 +429,328 @@ def test_create_tiled_regions_non_tma(randomize_setting, moly_run, moly_interval
+        assert ati in sample_auto_coords
+        assert sample_auto_coords[ati] == sample_auto_fovs_info[ati]
+
+        x, y = sample_auto_coords[ati]
+
+        # dark blue if row0_col0, else bright blue
+        if ati == 'row0_col0':
+            assert np.all(sample_slide_img[x, y, :] == np.array([50, 115, 229]))
+        else:
+            assert np.all(sample_slide_img[x, y, :] == np.array([162, 197, 255]))
+
+
+@pytest.mark.parametrize('randomize_setting', _REMAP_RANDOMIZE_TEST_CASES)
+@pytest.mark.parametrize('moly_insert', _REMAP_MOLY_INSERT_CASES)
+@pytest.mark.parametrize('moly_interval', _REMAP_MOLY_INTERVAL_CASES)
+def test_remap_and_reorder_fovs(randomize_setting, moly_insert, moly_interval):
+    # error check: moly_path must exist
+    with pytest.raises(FileNotFoundError):
+        tiling_utils.remap_and_reorder_fovs({}, {}, 'bad_path.json')
+
+    # define the sample Moly point
+    sample_moly_point = test_utils.generate_sample_fov_tiling_entry(
+        coord=(14540, -10830), name="MoQC"
+    )
+
+    # save the Moly point
+    with open('sample_moly_point.json', 'w') as smp:
+        json.dump(sample_moly_point, smp)
+
+    # error check: moly_interval must be at least 1
+    with pytest.raises(ValueError):
+        tiling_utils.remap_and_reorder_fovs({}, {}, 'sample_moly_point.json', moly_interval=0)
+
+    # define the coordinates and fov names manual by the user
+    manual_coords = [(0, 25), (50, 25), (50, 50), (75, 50), (100, 25), (100, 75)]
+    manual_fov_names = ['row%d_col%d' % (x, y) for (x, y) in manual_coords]
+
+    # generate the list of manual fovs
+    manual_sample_fovs = test_utils.generate_sample_fovs_list(
+        manual_coords, manual_fov_names
+    )
+
+    # define a sample mapping
+    sample_mapping = {
+        'row0_col25': 'row0_col0',
+        'row50_col25': 'row0_col25',
+        'row50_col50': 'row0_col100',
+        'row75_col50': 'row100_col100',
+        'row100_col25': 'row100_col0',
+        'row100_col75': 'row100_col75'
+    }
+
+    # copy the data so it doesn't overwrite manual_sample_fovs
+    manual_sample_fovs_copy = copy.deepcopy(manual_sample_fovs)
+
+    # add id, name, and status
+    manual_sample_fovs_copy['id'] = -1
+    manual_sample_fovs_copy['name'] = 'test'
+    manual_sample_fovs_copy['status'] = 'all_systems_go'
+
+    # remap the FOVs
+    new_manual_sample_fovs = tiling_utils.remap_and_reorder_fovs(
+        manual_sample_fovs_copy, sample_mapping, 'sample_moly_point.json', randomize_setting,
+        moly_insert, moly_interval
+    )
+
+    # assert id, name, and status are the same
+    assert new_manual_sample_fovs['id'] == manual_sample_fovs_copy['id']
+    assert new_manual_sample_fovs['name'] == manual_sample_fovs_copy['name']
+    assert new_manual_sample_fovs['status'] == manual_sample_fovs_copy['status']
+
+    # assert same number of FOVs exist in both original and scrambled (excluding Moly points)
+    assert len(manual_sample_fovs['fovs']) == \
```

Comment:
We can instead check that the `set` of FOV names between the two is equivalent except for Moly, right?

Reply:
Also change the comment to make it more clear what is being tested.

Reply:
Yup, we can do this by using `misc_utils.verify_same_elements`